systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -44,21 +47,47 @@ public static void waitForTopicStatus(String namespaceName, String topicPrefix,
 
     public static void waitForTopicStatus(String namespaceName, String topicPrefix, int numberOfTopics, Enum<?> conditionType, ConditionStatus conditionStatus) {
         LOGGER.info("Verifying that {} Topics are in {} state", numberOfTopics, conditionType.toString());
-        List<CompletableFuture<?>> topics = new ArrayList<>();
 
-        for (int i = 0; i < numberOfTopics; i++) {
-            String currentTopic = topicPrefix + i;
-            topics.add(CompletableFuture.runAsync(() ->
-                KafkaTopicUtils.waitForKafkaTopicStatus(namespaceName, currentTopic, conditionType, conditionStatus)
-            ));
+        // Determine the appropriate number of threads
+        int numberOfThreads = Math.min(Runtime.getRuntime().availableProcessors(), numberOfTopics);
+        ExecutorService customExecutor = Executors.newFixedThreadPool(numberOfThreads);
+
+        try {
+            List<CompletableFuture<?>> topics = new ArrayList<>();
+
+            for (int i = 0; i < numberOfTopics; i++) {
+                final String currentTopic = topicPrefix + i;
+                topics.add(CompletableFuture.runAsync(() ->
+                                KafkaTopicUtils.waitForKafkaTopicStatus(namespaceName, currentTopic, conditionType, conditionStatus),
+                        customExecutor // Use the custom executor
+                ));
+            }
+
+            CompletableFuture<Void> allTopics = CompletableFuture.allOf(topics.toArray(new CompletableFuture[0]))
+                    .thenRunAsync(() -> LOGGER.info("All Topics are in correct state"), customExecutor);
+
+            allTopics.join();
+        } finally {
+            // Ensure the executor shuts down even if an exception occurs
+            customExecutor.shutdown();
```

Comment:
how about `shutdownNow` here and save the one block?

Reply:
sure I can do it :) thanks